spec/rubocop/cop/performance/count_spec.rb

```diff
@@ -86,6 +86,20 @@
       expect(cop.highlights).to eq(["#{selector}(&:something).count"])
     end
 
+    it "registers an offense for #{selector}(&:something).count " \
+       'called on a class that includes or extends an enumerable' do
+      source = ['class A < Array',
+                '  def count(&block)',
+                "    #{selector}(&block).count",
+                '  end',
+                'end']
+      inspect_source(cop, source)
+
+      expect(cop.messages)
+        .to eq(["Use `count` instead of `#{selector}...count`."])
```

Comment:
This advise might trip some people into stackoverflow problems, but I guess that's preferable to an RuboCop error.